codegen/chain.go

```diff
@@ -637,12 +651,46 @@ func (cg *CodeGen) EmitFilesystemBuiltinChainStmt(ctx context.Context, scope *pa
 		}
 
 		fc = func(st llb.State) (llb.State, error) {
-			request, err := cg.outputRequest(ctx, st, Output{Type: OutputDockerPush, Ref: ref})
+			var digest string
+			request, err := cg.outputRequest(ctx, st, Output{Type: OutputDockerPush, Ref: ref},
+				solver.WithCallback(func(resp *client.SolveResponse) error {
+					digest = resp.ExporterResponse[keyContainerImageDigest]
+					return nil
+				}),
+			)
 			if err != nil {
 				return st, err
 			}
+
+			// XXX will this be de-duped if already solved by the binding?
```

Comment:
Yes, if it's not too complicated we should avoid duping but buildkit should dedupe for us.